docs/Middleware-and-Component.md

```diff
@@ -60,6 +60,49 @@ $middlewareStack->add(new AuthorizationMiddleware($this, [
 ]));
 ```
 
+### Handling unauthorized requests
+
+By default authorization exceptions thrown by the application are rethrown by the middleware.
+You can configure handlers for unauthorized requests and perform custom action, e.g.
+redirect the user to the login page.
+
+The built-in handlers are:
+
+* `Exception` - this handler will rethrow the exception, this is a default behavior of the middleware.
+* `Redirect` - this handler will redirect the request to the provided URL.
+* `CakeRedirect` - redirect handler with support for CakePHP Router.
+
+Both redirect handlers share the same configuration options:
+
+* `url` - URL to redirect to (`CakeRedirect` supports CakePHP Router syntax).
+* `exceptions` - a list of exception classes that should be redirected. By default only `MissingIdentityException` is redirected.
+* `queryParam` - the accessed request URL will be attached to the redirect URL query parameter (`url` by default).
+* `statusCode` - HTTP status code of a redirect, `302` by default.
+
+For example:
+
+```php
+$middlewareStack->add(new AuthorizationMiddleware($this, [
+    'unauthorizedHandler' => [
+        'className' => 'Authorization.Redirect',
+        'url' => '/users/login',
+        'queryParam' => 'redirectUrl',
+        'exceptions' => [
+            MissingIdentityException::class,
+            OtherException::class,
+        ],
+    ],
+]));
+```
+
+You can also add your own handler. Handlers should implement `Authorization\Middleware\Handler\HandlerInterface`,
+be suffixed with `Handler` suffix and reside under your app's or plugin's 
+`Middleware\Handler` namespace.
+
+Configuration options are passed to the handler's `handle()` method as the last parameter.
+
+Handlers catch only those exceptions which extend `Authorization\Exception\Exception` class.
```

Comment:
What about the whitelist? Could someone add `\RuntimeException` if they wanted?

Reply:
Catching non-authorization exceptions seems out of the scope of this plugin. More like a general exception handler thing.

Reply:
The code doesn't currently implement the check on `Authorization\Exception\Exception` though.

Reply:
Yes, this is intentional. I think that only missing identities should be redirected by default. Logged in users shouldn’t be redirected to the login page when they try to access unauthorized resource. This may cause a redirect loop. Also policy resolver exceptions shouldn’t be redirected as those are more like “internal error” exceptions.

Reply:
Makes sense.